### ISSUE TITLE
groupedservices doesn't display channel names

### ISSUE THREAD

**Issue by jeepcook** [openatv/enigma2#1889]
Hello,

Concerned github repository enigma2/data.

When the groupedservices file is correctly set, if you select a channel list in, you can display all sub channels defined in the groupedservices file by pressing green button.

But the only displayed information is the current epg for every sub channel. So if you have the same sub channel but with different versions, we are unable to know which one to select, because they have the same now epg, and no name. Same thing if you have a lot of distinct channels, because the channel name is not displayed you are unable to choice (example with the HD+ bouquet set in the default groupedservices file).

I think the solution could be to display "channel Name | now epg". For the channel name, the better solution could be to display the name of the channel set in the groupedservices files:

Example: for 2 french channels, their names are the same in lamedb (One is TF1, the second one FRANCE 2), so displaying the channel name extract from lamedb is not a solution because they are the same. But displaying the channel name set after the # could be great.

# TF1
1:0:19:2460:440:1:C00000:0:0:0:		#TF1 CSAT
1:0:19:601:6:20FA:DDE0000:0:0:0:	#TF1 PLS TNT
1:0:19:191:4FB0:55F:DDE0000:0:0:0:	#TF1 FRANSAT

# FRANCE 2
1:0:19:245A:440:1:C00000:0:0:0:		#FRANCE 2 CSAT
1:0:1:101:1:20FA:DDE0000:0:0:0:		#FRANCE 2 PLS TNT
1:0:19:12D:4F4C:55F:DDE0000:0:0:0:	#FRANCE 2 FRANSAT
.......

If you need some other informations, please contact me.

Thanks

**Comment by jeepcook** [comment]
Hello,
more and more, we have subversions of channels if we take n account iptv versions.
will this open issue  have a future?
In my example there are only lamedb channels but we can have iptv too. 
Please take in account this demand.
Thanks

**Comment by geeeeeeeeeee** [comment]
Is this a open Issue?
>  you can display all sub channels defined in the groupedservices file by pressing green button.

Can you provide a Screenshot?

**Comment by jeepcook** [comment]
Yes this issue is still open because nobody want to work on it.

Example, you have 3 times the same channel so they have the same program, instead of having channel name + epg , you will only have epg, do no way to know if for example it's the HD version, 4k version etc etc

**Comment by geeeeeeeeeee** [comment]
> If you need some other informations, please contact me.

>>  you can display all sub channels defined in the groupedservices file by pressing green button.

> Do you can provide a Screenshot?

**Comment by geeeeeeeeeee** [comment]
@jeepcook And?

**Comment by jbleyel** [comment]
Hi @jeepcook, what exactly do you mean with:

_"When the groupedservices file is correctly set.."_

Can you provide the file or bouquet entries where I can see what you mean.

**Comment by jbleyel** [comment]
I think I know what you mean.
The option channel section is working, but you have an issue with the EPG.
Right?

Please provide a screenshot from the screen where you wanna have more info.

**Comment by jeepcook** [comment]
Hi,
Sorry, I'm not at home, I'm at the hospital.
In fact an evolution from groupedservices functionality would be to display on screen the name of the channel exactly as defined after the # in the config file, name followed by the EPG.
I have no epg problem, but if for example I have 15 channels declinations but with the same EPG, I don't know how to select the channel I want to see, because the name is not displayed. This case is a good example for France 3 regional channels, same EPG but different local programs.
I hope I am clear.

Regards

**Comment by jbleyel** [comment]
This is what I see on my box:

<img width="592" alt="Bildschirm­foto 2023-01-07 um 12 42 15" src="https://user-images.githubusercontent.com/964927/211148836-680e3a08-ff62-44e3-8e4b-77e6dba4b602.png">

I see the different channel names and the EPG.

**Comment by jeepcook** [comment]
OK I forgot that this functionality was OK. But I have another case.
If I have 2 channel in a section of a config file, example TF1 and TF1 4K, the channel names are the same, the EPG too, but how to do the difference between them. It's the same case with a channel in HD, FHD, 4K etc . 
Other thing, the iptv isn't working. If you add 3 different channels with different locales (case usually existing in Pluto TV, samsung TV for example). The subservices functionality is broken.

**Comment by jbleyel** [comment]
If the channel name in lamedb is the same, than you have to change the bouquet file like this:

```
#SERVICE 1:0:19:2460:440:1:C00000:0:0:0:
#DESCRIPTION TF1 CSAT
#SERVICE 1:0:19:601:6:20FA:DDE0000:0:0:0:
#DESCRIPTION TF1 PLS TNT
#SERVICE 1:0:19:191:4FB0:55F:DDE0000:0:0:0:
#DESCRIPTION TF1 FRANSAT

```

**Comment by jeepcook** [comment]
Make this change in bouquet file is ok of course, but what about groupedservices?

**Comment by geeeeeeeeeee** [comment]
I did a fix, that it will work now with changed names in bouquet.

**Comment by jeepcook** [comment]
Ok thanks, I will try asap! Are the iptv urls ok too?

**Comment by geeeeeeeeeee** [comment]
> Are the iptv urls ok too?

It's your Issue, but a new one!

**Comment by jeepcook** [comment]
Ok thanks. So to display the specific channel name, I have to name them in bouquet. But any of them ?

**Comment by geeeeeeeeeee** [comment]
> But any of them ?
?

**Comment by jbleyel** [comment]
> Ok thanks. So to display the specific channel name, I have to name them in bouquet. But any of them ?

You need to do this only if the channel names are not unique in lamedb.

**Comment by jeepcook** [comment]
sorry. I have a lot of bouquet.tv files, so I have to name the channels in a specific bouquet.tv file or there are no rules so I can use "any of them", bouquet_gile1.tv, bouquet_file2.tv ...
Excuse if I make some mistakes in my english language!!

**Comment by geeeeeeeeeee** [comment]
You can do it in `bouquet_gile1.tv, bouquet_file2.tv ...`

**Comment by jeepcook** [comment]
great thanks.

**Comment by jeepcook** [comment]
Ok thanks, has this update been commited in OATV 7.1, 7.2?

**Comment by jbleyel** [comment]
What update?

**Comment by jeepcook** [comment]
This one from [geeeeeeeeeee](https://github.com/geeeeeeeeeee) 
"I did a fix, that it will work now with changed names in bouquet.
Nothing to do in groupedservices."

**Comment by jbleyel** [comment]
I have not seen any code.
As i said, you need to change your bouquet files.

**Comment by geeeeeeeeeee** [comment]
> Ok thanks, has this update been commited in OATV 7.1, 7.2?

Others are EOL and yes 7.x have this fix.

**Comment by geeeeeeeeeee** [comment]
You don't have 7.x?
Tested?

**Comment by jeepcook** [comment]
Yes I've now openatv 7.2. I have to do tests of subservices. Sorry I'm bored with personal problems.

**Comment by jeepcook** [comment]
Hello,
After years I'm back with this case. Same case with OATV 7.2 and 7.3.

Example:
I have in /etc/enigma2/groupedservices:
`#TF1
1:0:1:6359:6632:3256:1010101:0:0:0:http%3a//URLxxx/130803
1:0:1:AFD8:A0F7:F7D0:1010101:0:0:0:http%3a//URLxxx/131033
1:0:1:EAC9:E2A0:A0C1:1010101:0:0:0:http%3a//URLxxx/131034
#FRANCE 2
1:0:1:67BE:6F7A:7AB1:1010101:0:0:0:http%3a//URLxxx/130804
1:0:1:D7A6:DD7F:7FA9:1010101:0:0:0:http%3a//URLxxx/131035
1:0:1:406B:480F:F6B:1010101:0:0:0:http%3a//URLxxx/712245`

In the userbouquet file:
`#SERVICE 1:0:1:6359:6632:3256:1010101:0:0:0:http%3a//URLxxx/130803
#DESCRIPTION TF1 HEVC
#SERVICE 1:0:1:AFD8:A0F7:F7D0:1010101:0:0:0:http%3a//URLxxx/131033
#DESCRIPTION TF1 HD
#SERVICE 1:0:1:EAC9:E2A0:A0C1:1010101:0:0:0:http%3a//URLxxx/131034
#DESCRIPTION TF1 SD
#SERVICE 1:0:1:67BE:6F7A:7AB1:1010101:0:0:0:http%3a//URLxxx/130804
#DESCRIPTION FRANCE 2 HEVC
#SERVICE 1:0:1:D7A6:DD7F:7FA9:1010101:0:0:0:http%3a//URLxxx/131035
#DESCRIPTION FRANCE 2 HD
#SERVICE 1:0:1:406B:480F:F6B:1010101:0:0:0:http%3a//URLxxx/712245
#DESCRIPTION FRANCE 2 SD`
`
In the subservices screen I have this:
![1_0_1_6359_6632_3256_1010101_0_0_0_20230614211809](https://github.com/openatv/enigma2/assets/17811804/e5f9fa5a-f672-41fb-bd65-a4efe0d63475)

Instead of:
`0  TF1 HEVC  HH:MM-HH-MM EPG
1  TF1 HD HH:MM-HH-MM EPG
2 TF1 SD HH:MM-HH-MM EPG`

Another thing if ServiceRef is not in uppercase, the subservice menu doesn't work.

Hope somebody will read this old issue.

Thansks

**Comment by GhostofGeeeee** [comment]
> You can do it in `bouquet_gile1.tv, bouquet_file2.tv ...`

This was wrong.

I'm cognize the #DESCRIPTION doesn't work for this Screen. Like "all services" and wanted!
Maybe the URL is more unique/exact than #DESCRIPTION.

I'm out.

**Comment by jbleyel** [comment]
Hi @jeepcook ,

any news here?

**Comment by jbleyel** [comment]
Please try this:
[InfoBarGenerics.py.zip](https://github.com/openatv/enigma2/files/12266686/InfoBarGenerics.py.zip)

**Comment by jeepcook** [comment]
Hello,
Renamed /usr/lib/enigma2/python/Screens/InfoBarGenerics.pyc InfoBarGenerics.pyc.bak
cp new file to the same directory, restart enigma2 so InfoBarGenerics.pyc is generated

But what changes i'm supposed to see? In fact after accessing to subservices (green button) nothing has changed. Perhaps do you want I change something in my groupedservices file?

Thanks

**Comment by jeepcook** [comment]
view in the code that line has to be split by "#", it's the case on the channel I tested

**Comment by jbleyel** [comment]
Please change the userbouquet and groupservice.
You need to add : followed by the name at the end of the service line.

```
OLD:
#SERVICE 1:0:1:6359:6632:3256:1010101:0:0:0:http%3a//URLxxx/130803
#DESCRIPTION TF1 HEVC

NEW:
#SERVICE 1:0:1:6359:6632:3256:1010101:0:0:0:http%3a//URLxxx/130803:TF1 HEVC
#DESCRIPTION TF1 HEVC

```

**Comment by jeepcook** [comment]
I modified the .tv file:

#SERVICE 5002:0:1:7c4b:79c7:c742:1010101:0:0:0:http%3a//URLxxx/130803:TF1 HEVC
#DESCRIPTION TF1 HEVC
#SERVICE 5002:0:1:afd8:a0f7:f7d0:1010101:0:0:0:http%3a//URLxxx/131033:TF1 HD
#DESCRIPTION TF1 HD
#SERVICE 5002:0:1:eac9:e2a0:a0c1:1010101:0:0:0:http%3a//URLxxx/131034:TF1 SD
#DESCRIPTION TF1 SD
#SERVICE 5002:0:1:cabd:c9aa:aab4:1010101:0:0:0:http%3a//URLxxx/15892:TF1+1 FHD
#DESCRIPTION TF1+1 FHD
#SERVICE 5002:0:1:26cc:2863:63c6:1010101:0:0:0:http%3a//URLxxx/679562:TF1+1 HD
#DESCRIPTION TF1+1 HD

But the result is the same as the previous screenshot.

**Comment by jbleyel** [comment]
Why 5002:?

Please try this:

```
#SERVICE 1:0:1:7C4B:79C7:C742:1010101:0:0:0:http%3a//URLxxx/130803:TF1 HEVC
#DESCRIPTION TF1 HEVC

```

**Comment by jeepcook** [comment]
5002 because I'm using exteplayer3 for some streams, 5001 gstplayer and 4097 gstreamer for others. 1 is for satellite.

**Comment by jeepcook** [comment]
tested with 1 but same result

![image](https://github.com/openatv/enigma2/assets/17811804/6b5452e7-7432-4a64-9e78-aebd83460cf0)

**Comment by jbleyel** [comment]
You also need to change the groupservice file.
The sref must be identical.

I have also testet 4097:xxx and this is working here.

**Comment by jbleyel** [comment]
#SERVICE 4097:0:1:7C4B:79C7:C742:1010101:0:0:0:http%3a//URLxxx/130803:TF1 HEVC
#DESCRIPTION TF1 HEVC

groupservice

4097:0:1:7C4B:79C7:C742:1010101:0:0:0:http%3a//URLxxx/130803:TF1 HEVC
.
.

**Comment by jeepcook** [comment]
But you made another version? Because now for the channels I tested the subservices function is broken after make the changes in .tv file and groupedservices file.
![image](https://github.com/openatv/enigma2/assets/17811804/5376ea32-e50f-4714-9541-4e1aa768d3cc)

**Comment by jeepcook** [comment]
If we need to have the same syntax in.tv file and in groupedservices, why the previous system is still working and the new one not.  I miss a file?

**Comment by jbleyel** [comment]
I donno what you are doing:


This is my bouquet.

`
#SERVICE 4097:0:19:151A:455:1:C00000:0:0:0:http ...... :DMAX TEST
`

This is my groupservice

```
#TEST
1:0:19:283D:3FB:1:C00000:0:0:0:
4097:0:19:151A:455:1:C00000:0:0:0:http ...... :DMAX TEST

```

You can use the original InfoBarGenerics.

**Comment by jbleyel** [comment]
The sref in bouquet and in groupservice must be identical!

**Comment by jeepcook** [comment]
I made a lot of tests, I don't understand why I am unable to do working this. Could you please test the case when your group"#TEST" contains only ServiceRef+urls+Name please.
Thanks

**Comment by jbleyel** [comment]
[InfoBarGenerics.py.zip](https://github.com/openatv/enigma2/files/12267765/InfoBarGenerics.py.zip)

And you need the new ServiceInfo.py Converter.

[ServiceInfo.py.zip](https://github.com/openatv/enigma2/files/12267767/ServiceInfo.py.zip)

**Comment by jeepcook** [comment]
Thanks but enigma2 doesn't start anymore if I install new ServiceInfo.py.

**Comment by jeepcook** [comment]
Just with InfoBarGenerics.py there is a real evolution!
Now it works even if the .tv is not ending by ":NAME_OF_THE_CHANNEL". The subservice functionnalitu read the name in the groupedservices file.

![image](https://github.com/openatv/enigma2/assets/17811804/032c870f-ae30-41bf-bc3b-4e825e9ed48e)


Thanks

**Comment by jbleyel** [comment]
ServiceInfo.py is running here.
What's the error?

**Comment by jeepcook** [comment]
> ServiceInfo.py is running here. What's the error?

I had to restore old version , if not enigma2 doesn't start and bootloop.

**Comment by jbleyel** [comment]
Copy the new ServiceInfo.py and then input
```
init 4
enigma2

```
via ssh console or telnet

**Comment by jeepcook** [comment]
detect this error:

`20:02:40.5054 Traceback (most recent call last):
20:02:40.5056   File "/usr/lib/enigma2/python/StartEnigma.py", line 765, in <module>
20:02:40.5070     from Screens import InfoBar
20:02:40.5072   File "/usr/lib/enigma2/python/Screens/InfoBar.py", line 9, in <module>
20:02:40.5075   File "/usr/lib/enigma2/python/Screens/MovieSelection.py", line 34, in <module>
20:02:40.5077   File "/usr/lib/enigma2/python/Screens/ParentalControlSetup.py", line 8, in <module>
20:02:40.5079   File "/usr/lib/enigma2/python/Screens/ChannelSelection.py", line 46, in <module>
20:02:40.5081   File "/usr/lib/enigma2/python/Screens/ServiceInfo.py", line 3, in <module>
20:02:40.5085     from Screens.InfoBarGenerics import hasActiveSubservicesForCurrentChannel
20:02:40.5087   File "/usr/lib/enigma2/python/Screens/InfoBarGenerics.py", line 38, in <module>
20:02:40.5092     from Screens.ChannelSelection import ChannelSelection, PiPZapSelection, BouquetSelector, SilentBouquetSelector, EpgBouquetSelector, service_types_tv
20:02:40.5094 ImportError: cannot import name 'ChannelSelection' from partially initialized module 'Screens.ChannelSelection' (most likely due to a circular import) (/usr/lib/enigma2/python/Screens/ChannelSelection.pyc)
2`

**Comment by jbleyel** [comment]
ServiceInfo.py needs to copy to the Converter subfolder and not in Screens subfolder

**Comment by jeepcook** [comment]
Ok now installed in the Components/Converter folder. enigma2 start is OK :-) 
When I'm on a service the subservice logo is on, so green button, select another subservice, zap to it ok but subservice logo is now off and no subservices available :-(

**Comment by jbleyel** [comment]
> Ok now installed in the Components/Converter folder. enigma2 start is OK :-)
> When I'm on a service the subservice logo is on, so green button, select another subservice, zap to it ok but subservice logo is now off and no subservices available :-(

Strange

**Comment by jbleyel** [comment]
No issue here.

**Comment by jeepcook** [comment]
If I'm going on each channels included in a groupedservices group by a classic zap way, I have the logo on on each channel, and I can see the subservices list and zap on another subservice by this way. But after that the subservices functionnality is broken

**Comment by jbleyel** [comment]
Sorry. I cannot reproduce this.

**Comment by jeepcook** [comment]
> Sorry. I cannot reproduce this.

I don't understand. I will do other test to understand why I have the problem.
btw do you fix the uppercase bug? is groupedservices case sensitive?

**Comment by jbleyel** [comment]
> > Sorry. I cannot reproduce this.
> 
> I don't understand. I will do other test to understand why I have the problem. btw do you fix the uppercase bug? is groupedservices case sensitive?


There is no uppercase bug .. As I already said. The sref in bouquet file and groupedservices file must be identical.

**Comment by jeepcook** [comment]
I confirm the old bug with uppercase is still there. The subservices menu is only available if in groupedservices file the ServiceRef are uppercase. In lowercase the menu and so the icon are broken.
This is the first problem. The second one is that I see the subservices menu on each channel in the TV bouquet, but after select one of them the menu for the selected channel disapears
The third is that QuickZap generate a crash. See the log. 
Thanks

22:51:06.7126 Traceback (most recent call last):
22:51:06.7127   File "/usr/lib/enigma2/python/Components/ActionMap.py", line 305, in action
22:51:06.7219   File "/usr/lib/enigma2/python/Components/ActionMap.py", line 276, in action
22:51:06.7225   File "/usr/lib/enigma2/python/Screens/SubservicesQuickzap.py", line 66, in previousSubservice
22:51:06.7231 TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
22:51:06.7232 [ePyObject] (PyObject_CallObject(<bound method NumberActionMap.action of <Components.ActionMap.NumberActionMap object at 0xad4571d0>>,('DirectionActions', 'left')) failed)

[20230805-225106-enigma2-crash.log](https://github.com/openatv/enigma2/files/12268181/20230805-225106-enigma2-crash.log)

**Comment by jbleyel** [comment]
There is no case sensitive bug.
This is by design.
I’m not able to reproduce the second problem.
I will check the crash.

**Comment by jeepcook** [comment]
> There is no case sensitive bug

But just try, instead of uppercase in groupedservices file put ServiceRef in lowercase, you will see the subservices are broken for the group. This bug has always been there. For me it's a bug.

**Comment by jbleyel** [comment]
Please read my comments.
The sRef must be identical.
It’s case sensitive and I won’t change this because this cause other issues.

**Comment by jeepcook** [comment]
I have identical sRef. If they are in lowercase the subservices does not appear, and in uppercase yes. So there is a real problem. You just have to test it, you will see. I'm not crazy...

**Comment by jbleyel** [comment]
Is the sRef lowercase in lamedb ? NO!
That’s your misunderstanding.

**Comment by jbleyel** [comment]
The first part of the sRef is uppercase since years.

**Comment by jeepcook** [comment]
With iptv lamedb isn't used anymore. But in this case don't say sRef must be identical. The real text is the sRef must be identical AND in uppercase.
The previous version worked better than the new which don't work. But you preferred commit it before the tests were finished.

**Comment by jbleyel** [comment]
I need your files 
groupservice
All bouquet files 
lamedb
epg.dat

otherwise I’m out

**Comment by jeepcook** [comment]
you've got mail ;-)

**Comment by jbleyel** [comment]
[ServiceInfo.py.zip](https://github.com/openatv/enigma2/files/12269505/ServiceInfo.py.zip)
[SubservicesQuickzap.py.zip](https://github.com/openatv/enigma2/files/12269543/SubservicesQuickzap.py.zip)
[InfoBarGenerics.py.zip](https://github.com/openatv/enigma2/files/12269556/InfoBarGenerics.py.zip)

**Comment by jeepcook** [comment]
Hi @jbleyel ,

Tests done on 10 channels and their subservices. 
Yeah, you did it, you fix it. It's a great news for me, this was an old issue. Everything is ok, quickzap included.
Will now configure my groupedservices file for all channels.
Thanks :-)

**Comment by jeepcook** [comment]
Hello,
When will this fix/update be merged to enigma2 7.3?
Thanks

**Comment by jbleyel** [comment]
Today
